api/app/Http/Middleware/TransformInputDataMiddleware.php

```diff
@@ -27,8 +27,11 @@ public function handle($request, Closure $next)
             // Find any potential camelCase keys in the 'root' array, and convert
             // them to snake_case
             if (!ctype_lower($key)) {
-                $request->offsetSet(snake_case($key), $value);
-                $request->offsetUnset($key);
+                // Only convert if the key will change
+                if ($key != snake_case($key)) {
```

Comment:
@xiphiaz Keys from external services. If we proxy them through the frontend as planned, I suppose the frontend might convert the variables returned from the provider to camelCase. But for me that have received the responses directly from the providers, receive responses in snake_case, like `oauth_token`, and in that case it didn't convert it, as it was already in snake_case, but unset it after "conversion" to the variables got removed from the request for the application to process.